integration_tests/src/main/python/window_function_test.py

```diff
@@ -240,6 +246,36 @@ def test_decimal_running_sum_window_no_part(data_gen):
         'from window_agg_table',
         conf = {'spark.rapids.sql.batchSizeBytes': '100'})
 
+@ignore_order
+@approximate_float
+@pytest.mark.parametrize('data_gen', numeric_gens, ids=idfn)
+def test_numeric_running_sum_window_no_part_unbounded(data_gen):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark: two_col_df(spark, UniqueLongGen(), data_gen),
+        'window_agg_table',
+        'select '
+        ' sum(b) over '
+        '   (order by a asc '
+        '      rows between UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) as sum_b_asc '
+        'from window_agg_table',
+        conf = {'spark.rapids.sql.variableFloatAgg.enabled': 'true',
+                'spark.rapids.sql.batchSizeBytes': '100'})
+
+@ignore_order
+@approximate_float
+@pytest.mark.parametrize('data_gen', numeric_gens, ids=idfn)
+def test_numeric_running_sum_window_no_part_unbounded_partitioned(data_gen):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark: two_col_df(spark, UniqueLongGen(), data_gen).repartition(256),
```

Comment:
I don't think increasing the partitions is going to trigger the previous error. To trigger the issue the `previousValue` has to be `None`, which can happen the first time processing is started or if an overflow happened for decimal. But also the `samePartitionMask` must be either `scala.Left(cv)` or `scala.Right(true)`. In the case we saw it was the latter one.

That indicates that we hit an overflow when doing decimal processing (which again matches with the negative scale), and the next entire batch is for the same grouping (Meaning there is no partition by key, only an order by key) which these tests cover.

I think you are more likely to hit the problem by having more rows with very large decimal values in them instead of trying to partition the data more.

Reply:
If I revert the fix for the scale, I do see this failure:

```
23/08/23 17:58:11 WARN TaskSetManager: Lost task 0.0 in stage 31.0 (TID 79) (192.168.0.80 executor driver): java.lang.AssertionError: Type conversion is not allowed from Table{columns=[ColumnVector{rows=598, type=DECIMAL128 scale:-1, nullCount=Optional.empty, offHeap=(ID: 144448 7f905c03d4a0)}, ColumnVector{rows=598, type=INT64, nullCount=Optional.empty, offHeap=(ID: 144449 7f905c2334a0)}, ColumnVector{rows=598, type=DECIMAL64 scale:0, nullCount=Optional.empty, offHeap=(ID: 144995 7f905c039810)}, ColumnVector{rows=598, type=DECIMAL128 scale:1, nullCount=Optional.empty, offHeap=(ID: 144997 7f905c20b270)}], cudfTable=140257995911808, rows=598} to [DecimalType(38,1), LongType, DecimalType(10,0), DecimalType(38,1)] columns 0 to 4
```

Reply:
> I think you are more likely to hit the problem by having more rows with very large decimal values in them instead of trying to partition the data more.

How would I achieve this?

Reply:
I tried to reproduce it and couldn't I'm not really sure. Would need to do some more debugging to see why I am not hitting it when I think I should.